pkgs/by-name/va/vala-panel-appmenu/proper-install-dirs.patch

```diff
@@ -0,0 +1,95 @@
+diff --git a/applets/meson.build b/applets/meson.build
+index fd29a9f..56f89dd 100644
+--- a/applets/meson.build
++++ b/applets/meson.build
+@@ -4,7 +4,6 @@ if vala_panel_found
+     vpp = shared_module('appmenu', vp_sources, libres,
+                     dependencies: [appmenu_dep, vp],
+                     install: true,
+-                    install_dir: vp_applets_libdir
+                   )
+ endif
+ 
+@@ -14,7 +13,6 @@ if mate_found
+     mpp = shared_module('appmenu-mate', mp_sources, libres,
+                     dependencies: [appmenu_dep, mp],
+                     install: true,
+-                    install_dir: mate_applet_libdir
+                   )
+ endif
+ 
+@@ -24,7 +22,6 @@ if xfce_found
+     xpp = shared_module('appmenu-xfce', xp_sources, libres,
+                     dependencies: [appmenu_dep, xc,xp],
+                     install: true,
+-                    install_dir: xfce_applet_libdir
+                   )
+ endif
+ 
+@@ -34,6 +31,5 @@ if budgie_found
+     bpp = shared_module('appmenu-budgie', bp_sources, libres,
+                     dependencies: [appmenu_dep, bp],
+                     install: true,
+-                    install_dir: budgie_applet_libdir
+                   )
+ endif
+diff --git a/data/meson.build b/data/meson.build
+index fa3df0c..56b327e 100644
+--- a/data/meson.build
++++ b/data/meson.build
+@@ -3,7 +3,7 @@ cdata.set('CMAKE_INSTALL_FULL_LIBDIR', join_paths(prefix,get_option('libdir')))
+ cdata.set('CMAKE_INSTALL_FULL_DATAROOTDIR', join_paths(prefix,datadir))
+ 
+ if vala_panel_found
+-    vp_applets_datadir = vp.get_variable(pkgconfig:'applets_data')
++    vp_applets_datadir = get_option('datadir') / 'applications'
```

Comment:
By the way, you can override pkg-config variables with environment variables:

https://gitlab.freedesktop.org/pkg-config/pkg-config/-/blob/d97db4fae4c1cd099b506970b285dc2afd818ea2/pkg-config.1#L378-383

Reply:
It tries to write into another package’s tree because the above line essentially calls `pkg-config vala-panel --variable applets_data`, which will look into `vala-panel.pc` and see that project’s datadir. It works on other distros because they install everything to `/usr` but since Nix ditched [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard), we are not so lucky.

Ideally, the project would use [`pkgconfig_define`](https://mesonbuild.com/Reference-manual_returned_dep.html#get_variable_pkgconfig_define) argument to override the `datadir` to its own as described in https://www.bassi.io/articles/2018/03/15/pkg-config-and-paths/ but patching it or overriding it in Nixpkgs will be necessary until then.

Reply:
I am a little bit confused, should I apply your suggestion or we can ignore this?

Reply:
I am suggesting that the preferred solution would be opening a merge request against https://gitlab.com/vala-panel-project/vala-panel-appmenu to use `pkgconfig_define` since then we would not need anything special in Nixpkgs. See https://gitlab.freedesktop.org/pipewire/pipewire/-/commit/6b06341b1465270d784bd68bdc82156b687b8930 for an example.

But if you do not feel like it, the current patch should be fine (after fixing the paths).